src/nvim/misc1.c

```diff
@@ -1292,11 +1293,12 @@ int plines_win_nofold(win_T *wp, linenr_T lnum)
   width = wp->w_width - win_col_off(wp);
   if (width <= 0)
     return 32000;
-  if (col <= width)
+  if (col <= (unsigned int)width)
     return 1;
   col -= width;
   width += win_col_off2(wp);
-  return (col + (width - 1)) / width + 1;
+  assert(col <= INT_MAX);
```

Comment:
Ah, ok. Now I get you. 
A couple of gotchas with the proposed assert:
- `(int)col` would be unsafe without first checking col < INT_MAX.
- We are doing calculations as `int`, not `int32_t`. So, it should be INT_MAX, not INT32_MAX.

So, it would remain:
```
assert(col <= INT_MAX && (int)col < INT_MAX - (width - 1));
```
Agree?